tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -482,12 +494,109 @@ func (az *AzureProvider) constructStorageProfile(imageId string) (storageProfile
 }
 
 // getWorkerBackendPoolID gets the backend pool id of the worker node loadbalancer
-func (az *AzureProvider) getWorkerBackendPoolID() *string {
-	// We assume that BackendPoolID for worker vm follows the following format
-	// /subscriptions/$SUBSCRIPTID/resourceGroups/$INFRAID-rg/providers/Microsoft.Network/loadBalancers/$INFRAID/backendAddressPools/$INFRAID
-	var wbpi = fmt.Sprintf("/subscriptions/%s/resourceGroups/%[2]s-rg/providers/"+
-		"Microsoft.Network/loadBalancers/%[2]s/backendAddressPools/%[2]s", az.subscriptionID, az.infraID)
-	return &wbpi
+func (az *AzureProvider) getWorkerBackendAddressPools(ctx context.Context) ([]network.BackendAddressPool, error) {
+	// Create a filter to get a resource of type virtualMachine which contains 'worker' in its name
+	filter := "resourceType eq 'Microsoft.Compute/virtualMachines' and substringof('worker', name)"
+	// number of resources to retrieve
+	// https://learnk8s.io/kubernetes-node-size
+	// Kubernetes can technically accommodate 5000 nodes, considering that for optimal performance organizations limit at 500 worker nodes
+	top := int32(500)
+	// Use getResource() function to get all the workerVMs by filtering VMs which have 'worker' in their name
+	vmResources, err := az.getResource(ctx, filter, top)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get worker node: %v", err)
+	}
+	if vmResources == nil {
+		return nil, fmt.Errorf("failed to get worker node VM resource")
+	}
+
+	// create a map to keep track of all the loadbalancers associated with different worker nodes.
+	lbSet := make(map[string]struct{})
+	// filler value for map
+	var exists = struct{}{}
+
+	// Iterate over all the VM resources and get the worker backend pools attached to all the resources.
+	for _, vmResource := range vmResources {
+		vmName := extractResourceName(*vmResource.ID)
+		err, nicName := az.getNICname(ctx, vmName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get NIC name %v", err)
+		}
+
+		interfaceStruct, err := az.nicClient.Get(ctx, az.resourceGroupName, nicName, "")
+		if err != nil {
+			return nil, fmt.Errorf("cannot fetch the network interface data of %s: %v", vmName, err)
+		}
+
+		if interfaceStruct.InterfacePropertiesFormat == nil {
+			return nil, fmt.Errorf("ip configuration for %s cannot be nil", vmName)
+		}
+		interfacePropFormat := *(interfaceStruct.InterfacePropertiesFormat)
+
+		if interfacePropFormat.IPConfigurations == nil {
+			return nil, fmt.Errorf("ip configuration for %s cannot be nil", vmName)
+		}
+		interfaceIPConfigs := *(interfacePropFormat.IPConfigurations)
+
+		if interfaceIPConfigs == nil {
+			return nil, fmt.Errorf("ip configuration properties for %s cannot be nil", vmName)
+		}
+		// we assume that all worker nodes have only one IP config.
+		// if the worker nodes have more than one IP config, throw an error
+		if len(interfaceIPConfigs) > 1 {
+			return nil, fmt.Errorf("multiple interface ip configs are not supported for VM %s", vmName)
```

Comment:
`Existing worker `VM %s has multiple NICs. This is not supported by WNI`. Or something to do that effect.